server/api/auth/login.post.ts

```diff
@@ -0,0 +1,86 @@
+import * as v from "valibot"
+import bcrypt from "bcrypt"
+import User from "~~/server/database/models/User"
+
+const schema = v.object({
+  usernameOrEmail: v.pipe(
+    v.string(),
+    v.nonEmpty("Please enter your email."),
+  ),
+  password: v.string(),
+})
+
+const error = createError({
+  status: 401,
+  message: "Username, email or password wrong",
+})
+
+export default defineEventHandler(async (event) => {
+  const c = await readValidatedBody(event, body => v.parse(schema, body))
+
+  const whereClause
+    = v.is(v.pipe(v.string(), v.email()), c.usernameOrEmail)
```

Comment:
Maybe just use a (local) helper function isEmail then this is more readable and the ternary operator is smaller and teamscale should accept it like it did in legal